cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -228,17 +240,34 @@ func upgradeComponent(component string, certsRenewMgr *renewal.Manager, waiter a
 		}
 	}
 
-	// Move the old manifest into the old-manifests directory
-	if err := pathMgr.MoveFile(currentManifestPath, backupManifestPath); err != nil {
-		return rollbackOldManifests(recoverManifests, err, pathMgr, recoverEtcd)
-	}
+	// This can only happen if the component is etcd, cert renewal is enabled and kube-apiserver needs an upgrade
+	if equal && certsRenewMgr != nil {
+		fmt.Printf("[upgrade/staticpods] Restarting the %s static pod and backing up its manifest to %q\n",
+			component, backupManifestPath)
 
-	// Move the new manifest into the manifests directory
-	if err := pathMgr.MoveFile(newManifestPath, currentManifestPath); err != nil {
-		return rollbackOldManifests(recoverManifests, err, pathMgr, recoverEtcd)
-	}
+		// Move the old manifest into the old-manifests directory
+		if err := pathMgr.MoveFile(currentManifestPath, backupManifestPath); err != nil {
```

Comment:
having this second move seems sufficient to perform a quick pod restart of etcd it seems.

i checked with crictl ps multiple times;
- old container exits
- new container starts

i also tested by adding a few seconds timeout here (time.Sleep), but it resulted in a slow etcd pod restart and there was 20+ seconds downtime of e.g. `k get po -A`  not getting a response.

if for some reason the etcd pod does not properly restart here, once its serving cert expires the health check will restart it after 15 seconds. i could not reproduce such a scenario, because it always restarts after the two "moves" here.

Reply:
updated. i missed this optimization because of WIP refactors when i was working on the PR